pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -98,6 +98,8 @@ func NewSnapshotter(logger *logrus.Entry, config *Config) *Snapshotter {
 		deltaSnapshotReqCh: make(chan struct{}),
 		fullSnapshotAckCh:  make(chan error),
 		deltaSnapshotAckCh: make(chan error),
+		fullSnapshotCh:     make(chan *snapstore.Snapshot),
```

Comment:
Will you please combine fullSnapshstoCh and FullSnapshtoAckCh into single channel FullSnapshtoAckCh of struct something like
```
snapshotter.result{
snapshot.Snapshot : snapshot,
error: err
}
```
Same for deltaSnaphostCh.